pkg/endpoint/endpoint.go

```diff
@@ -385,6 +403,9 @@ type Endpoint struct {
 	// mutable! must hold the endpoint lock to read
 	ciliumEndpointUID k8sTypes.UID
 
+	// metadata is used to store some internal metadata about this Endpoint.
+	metadata map[string]interface{}
```

Comment:
naming nit: this isn't metadata, it's data. In other words, it directly influences the endpoint's configuration, same as anything else. Plus, metadata already implicitly means name + namespace in the k8s world.

What about calling this TestTweaks? Or something like that?

Reply:
Or maybe `properties`?

Reply:
Or maybe `properties`?